### PR TITLE
Consume common java as implementation versus sourceDirs

### PR DESCRIPTION
**Pull request by shahzaibj**
Consuming as sourceDirs was hiding problems...so we should just consume as implementation.

Basically the common-java project was using JAVA 8 and that would cause common to not build. In this PR, I have explicitly set source compatibility of common java project to JAVA 7 (we will change this to 8 when we have actually decided to do it).

Also applied the plugin so we can force set to JAVA 8 when running tests as we already have written tests that are using JAVA 8

Related PR: https://github.com/AzureAD/android-complete/pull/78

### CODE REVIEW

**Review by AdamBJohnsonx — APPROVED** [review]

**Comment by AdamBJohnsonx** [comment]
The instrumented tests might run if we fix duplicate class error in: modules jetified-jcip-annotations-1.0 (net.jcip:jcip-annotations:1.0) and jetified-jcip-annotations-1.0-1 (com.github.stephenc.jcip:jcip-annotations:1.0-1)

**Review by rpdome — APPROVED** [review]
Yeah. This works. Not sure why I didn't think of it 😅 

Thanks Shahzaib!

**Comment by shahzaibj** [comment]
> Yeah. This works. Not sure why I didn't think of it 😅
> 
> Thanks Shahzaib!

@rpdome I didn't get it fully working yet. 😁 Getting this when building MSAL Test App:

`E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.msft.identity.client.sample.local, PID: 13838
    java.lang.NoClassDefFoundError: Failed resolution of: Lcom/microsoft/identity/common/java/logging/ILoggerCallback;
        at com.microsoft.identity.common.internal.logging.Logger.getInstance(Logger.java:223)
        at com.microsoft.identity.client.Logger.setExternalLogger(Logger.java:147)
        at com.microsoft.identity.client.testapp.MsalSampleApp.onCreate(MsalSampleApp.java:54)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1189)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6460)
        at android.app.ActivityThread.access$1300(ActivityThread.java:219)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.microsoft.identity.common.java.logging.ILoggerCallback" on path: DexPathList[[zip file "/data/app/com.msft.identity.client.sample.local-jRljr5Q2IpVgDANW-xvT2g==/base.apk"],nativeLibraryDirectories=[/data/app/com.msft.identity.client.sample.local-jRljr5Q2IpVgDANW-xvT2g==/lib/arm64, /system/lib64, /product/lib64]]
`

File: common4j/build.gradle [hunk 1]
```diff
@@ -9,6 +9,17 @@
 plugins {
     // Apply the java-library plugin for API and implementation separation.
     id 'java-library'
+    id 'com.microsoft.identity.buildsystem' version '0.0.1'
```
Comment (iamgusain) [inline common4j/build.gradle:12]: >0.0.1 [](http://example.com/codeflow?start=53&length=5)

can change to 0.1.0 now..

**Review by iamgusain — APPROVED** [review]
:shipit:

File: common4j/build.gradle (hunk 1 above)
Comment (shahzaibj) [reply common4j/build.gradle]: Done

**Pull request merged**